### PR TITLE
Add missing tests for dynamically loading matchers, simplify plugin design

### PR DESCRIPTION
**Pull request by krzkaczor**
- [x] missing tests for dynamically loaded matchers/validators
- [x] simplify plugin design to `PluginConfig` load
- [x] derive plugin typings automatically 
- [x] update `CONTRIBUTING.md` with monorepo information
- [x] Add missing changelogs

### CODE REVIEW

File: packages/earljs/src/plugins/types.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import { Expectation } from '../Expectation'
-import { Matcher } from '../matchers/Base'
 import { SmartEqResult } from '../validators/smartEq'
 
 export type DynamicValidator<T> = (this: Expectation<T>, ...args: any[]) => void | Promise<void>
-export type DynamicMatcher = (...args: any[]) => Matcher
+export type DynamicMatcher = (...args: any[]) => any // real type should be Matcher but can be casted to anything for improved DX
```
Comment (sz-piotr) [inline packages/earljs/src/plugins/types.ts:5]: casted -> cast

File: packages/example-plugin/types.d.ts [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-    evenNumber: typeof EvenNumberMatcher.make
-  }
+type Config = typeof numberUtilsPlugin
+type Matchers = Config['matchers']
+type Validators = Config['validators']
 
-  interface Expectation<T> {
-    toBeEven: typeof toBeEven
-  }
+declare module 'earljs' {
```
Comment (sz-piotr) [inline packages/example-plugin/types.d.ts:7]: Nice!

File: packages/example/test/NumberUtilsPlugin.test.ts [hunk 3]
```diff
@@ -3,6 +3,9 @@ import { expect } from 'earljs'
 describe('NumberUtilsPlugin', () => {
   it('EvenNumberMatcher works', () => {
     expect(2).toEqual(expect.evenNumber())
+
+    // it's typesafe! this is a compile time error
+    // expect('2').toEqual(expect.evenNumber())
```
Comment (sz-piotr) [inline packages/example/test/NumberUtilsPlugin.test.ts:8]: Isn't there a `@ts` annotation that will check that this is an error?

**Review by sz-piotr — CHANGES_REQUESTED** [review]

File: packages/example/test/NumberUtilsPlugin.test.ts (hunk 3 above)
Comment (krzkaczor) [reply packages/example/test/NumberUtilsPlugin.test.ts:8]: Uhh yeah i even used it at the beginning but then realized I don't want this test to fail... I ll add another test case for this.

**Pull request merged**